src/views/project-panel/project-panel.tsx

```diff
@@ -199,8 +197,15 @@ export const ProjectPanel = withStyles(styles)(
 
             handleContextMenu = (event: React.MouseEvent<HTMLElement>, resourceUuid: string) => {
                 const kind = resourceKindToContextMenuKind(resourceUuid);
-                if (kind) {
-                    this.props.dispatch<any>(openContextMenu(event, { name: '', uuid: resourceUuid, kind }));
+                const resource = getResource(resourceUuid)(this.props.resources) as ProjectResource;
```

Comment:
Could you provide ProjectResource as a generic type parameter to getResource call?